### PR TITLE
changed context canceled to 499

### PR DESCRIPTION
**Pull request by juliaElastic**
## What is the problem this PR solves?

// Please do not just reference an issue. Explain WHAT the problem this PR solves here.

SLOs alerts are being triggered because of false 503 errors.

## How does this PR solve the problem?

// Explain HOW you solved the problem in your code. It is possible that during PR reviews this changes and then this section should be updated.

Change context canceled errors to 499, like the proxy does it. 

## How to test this PR locally

Enroll an agent and terminate the process during checkin/enroll. Verify that a 499 error is logged.

## Design Checklist

- [ ] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.
- [ ] I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.
- [ ] I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/fleet-server#changelog)

## Related issues

Closes https://github.com/elastic/fleet-server/issues/3010

### CODE REVIEW

**Comment by juliaElastic** [comment]
Tested locally by enrolling an agent and terminating the command before it succeeds.

```
09:39:09.219 ERR Dispatch abort response error.message="context canceled" action=create ecs.version=1.6.0 mod=bulk refresh=true rtt=392.509459 service.name=fleet-server service.type=fleet-server
09:39:09.219 WRN HTTP request error error.message="context canceled" ecs.version=1.6.0 event.duration=452986000 http.request.id=01HBZK93APJ9F7P4TEDZ4MEDTA http.response.status_code=499 server.address= service.name=fleet-server service.type=fleet-server
09:39:09.220 ERR perform rollback on enrollment failure error.message="context canceled" ecs.version=1.6.0 http.request.id=01HBZK93APJ9F7P4TEDZ4MEDTA mod=enroll server.address= service.name=fleet-server service.type=fleet-server
09:39:09.220 DBG rollback function called ecs.version=1.6.0 http.request.id=01HBZK93APJ9F7P4TEDZ4MEDTA mod=enroll rollback_fn_name="invalidate API key" server.address= service.name=fleet-server service.type=fleet-server
09:39:09.220 ERR Fail ApiKeyRead error.message="request to elasticsearch failed: context canceled" ecs.version=1.6.0 fleet.apikey.id=GzI0_4oB3ZNBSE2MjXNj fleet.enroll.apikey.id=Ppf99YoB93mw0XwOGtMk http.request.id=01HBZK93APJ9F7P4TEDZ4MEDTA mod=enroll server.address= service.name=fleet-server service.type=fleet-server
09:39:09.220 ERR rollback function "invalidate API key" failed error.message="request to elasticsearch failed: context canceled" ecs.version=1.6.0 http.request.id=01HBZK93APJ9F7P4TEDZ4MEDTA mod=enroll rollback_fn_name="invalidate API key" server.address= service.name=fleet-server service.type=fleet-server
09:39:09.220 ERR rollback error on enrollment failure error.message="request to elasticsearch failed: context canceled" ecs.version=1.6.0 http.request.id=01HBZK93APJ9F7P4TEDZ4MEDTA mod=enroll server.address= service.name=fleet-server service.type=fleet-server
09:39:09.220 INF 499 HTTP Request client.address=192.168.178.216:59411 ecs.version=1.6.0 event.duration=454014000 fleet.apikey.id=Ppf99YoB93mw0XwOGtMk http.request.body.bytes=643 http.request.id=01HBZK93APJ9F7P4TEDZ4MEDTA http.request.method=POST http.response.body.bytes=85 http.response.status_code=499 http.version=1.1 server.address= service.name=fleet-server service.type=fleet-server tls.established=false url.full=/api/fleet/agents/enroll
```

File: internal/pkg/api/error.go [hunk 1]
```diff
@@ -74,10 +74,10 @@ func NewHTTPErrResp(err error) HTTPErrResp {
 		{
 			context.Canceled,
 			HTTPErrResp{
-				http.StatusServiceUnavailable,
-				"ServiceUnavailable",
+				http.StatusClientClosedRequest,
+				"StatusClientClosedRequest",
 				"server is stopping",
-				zerolog.DebugLevel,
+				zerolog.WarnLevel,
```
Comment (joshdover) [inline internal/pkg/api/error.go:80]: If we are sure this only will happen in cases of client disconnects, maybe info is fine

Comment (juliaElastic) [reply internal/pkg/api/error.go]: updated

**Review by joshdover — APPROVED** [review]

**Review by kpollich — APPROVED** [review]
LGTM 🚀

**Review by nchaulet — APPROVED** [review]
🚀 I am wondering if it something we can add a test for, so we do not remove it by error maybe in here https://github.com/michel-laterman/fleet-server/blob/e7c9722756d52496a2b0350e7a956df46f22b7af/testing/e2e/api_version/client_api_current.go#L6

**Pull request merged**